### PR TITLE
HDDS-6576. [Multi-Tenant] Update documentation around Ranger policy creation on bucket sharing

### PR DESCRIPTION
**Pull request by smengcl**
## What changes were proposed in this pull request?

Update documentation around Ranger policy creation on bucket sharing.

See the PR content for details. Should be self-explanatory. :)

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-6576

## How was this patch tested?

n/a

### CODE REVIEW

File: hadoop-hdds/docs/content/feature/S3-Multi-Tenancy-Access-Control.md [hunk 1]
```diff
@@ -66,3 +67,17 @@ The Ranger Sync thread does the following:
...
+
+
+## Sharing a bucket
+
+By default only the bucket owners have full access to the buckets they created.
+So in order to share a bucket with other users, a cluster admin or tenant admin will needs to manually create a new Ozone policy in Ranger for that bucket.  
+
+Further, if a cluster admin or tenant admin wants the bucket owner (who is a regular tenant user without any superuser privileges) to be able to edit that bucket's policy,
+when manually creating a new Ozone policy in Ranger for that bucket,
+an admin will need to explicitly grant the bucket owner user (note: specify an actual user name, `{OWNER}` tag should not be used here) ALL permission
```
Comment (errose28) [inline hadoop-hdds/docs/content/feature/S3-Multi-Tenancy-Access-Control.md:79]: Can we add a quick explanation why {OWNER} cannot be used?

File: hadoop-hdds/docs/content/feature/S3-Multi-Tenancy-Access-Control.md [hunk 2]
```diff
@@ -66,3 +67,17 @@ The Ranger Sync thread does the following:
 2. Checks if default tenant roles are out-of-sync (could be caused by OM crash during user assign/revoke operation). Overwrites them if this is the case.
 3. Performs all Ranger update (write) operations queued by Ozone tenant commands from the last sync, if any.
    - This implies there will be a delay before Ranger policies and roles are updated for any tenant write operations (tenant create/delete, tenant user assign/revoke/assignadmin/revokeadmin, etc.). 
+
+
+## Sharing a bucket
+
+By default only the bucket owners have full access to the buckets they created.
+So in order to share a bucket with other users, a cluster admin or tenant admin will needs to manually create a new Ozone policy in Ranger for that bucket.
```
Comment (errose28) [inline hadoop-hdds/docs/content/feature/S3-Multi-Tenancy-Access-Control.md:75]: Have we tested this process? Is the bucket policy going to conflict with the bucket policy being created as part of the tenant?

**Review by errose28 — COMMENTED** [review]
Thanks for the doc update @smengcl. I haven't worked much in this area of multi-tenancy so I have a few questions.

File: hadoop-hdds/docs/content/feature/S3-Multi-Tenancy-Access-Control.md (hunk 1 above)
Comment (smengcl) [reply hadoop-hdds/docs/content/feature/S3-Multi-Tenancy-Access-Control.md:79]: Sure. 'cause `{OWNER}` tag doesn't work with Ranger Web UI's "Delegated Admin" flag.

The `{OWNER}` tag is **only** meaningful when OM is doing a permission (ACL) check, in the process of which OM fills in what this `{OWNER}` tag actually stands for (e.g. user `hive` when the operation is bucket list and `hive` is the bucket owner, plus another volume read permission check on the with `{OWNER}` equals `om` (say `om` is the volume owner) before this).

This is also the exact reason why an admin might want to add this separate Ranger bucket policy here in the first place (in order for the bucket owner to be able to **edit** this newly added policy on their own), as we already have the default bucket policy with the `{OWNER}` tag having `ALL` permission on the policy. If they don't have such use case, they don't need to add this new policy for each bucket.

File: hadoop-hdds/docs/content/feature/S3-Multi-Tenancy-Access-Control.md (hunk 2 above)
Comment (smengcl) [reply hadoop-hdds/docs/content/feature/S3-Multi-Tenancy-Access-Control.md:75]: Good question. AFAIK resource denial takes precedence over any other allow rules, as depicted in this flow chart: https://docs.cloudera.com/cdp-private-cloud-base/7.1.7/security-ranger-authorization/topics/security-ranger-tags-and-policy-evaluation.html

Comment (errose28) [reply hadoop-hdds/docs/content/feature/S3-Multi-Tenancy-Access-Control.md]: We tested this and it works without conflict.

**Review by errose28 — APPROVED** [review]

**Comment by smengcl** [comment]
Thanks @errose28 for the review!

**Pull request merged**